packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -42,6 +43,35 @@ const Map<String, _HardwareType> _knownHardware = <String, _HardwareType>{
   'samsungexynos9810': _HardwareType.physical,
 };
 
+bool allowHeapCorruptionOnWindows(int exitCode) {
+  // In platform tools 29.0.0 adb.exe seems to be ending with this heap
+  // corruption error code onseemingly successful termination.
+  // So we ignore this error on Windows.
+  return exitCode == -1073740940 && platform.isWindows;
+}
+
+String runMostlyCheckedSync(
```

Comment:
These wrappers are only used to run adb, so how about renaming something like 'runAdbSync' and 'runAdbAsync'. You could also fold in the otherwise repeated calls to adbCommandForDevice